### PR TITLE
Upgrade specta to rc.20

### PR DESCRIPTION
**Pull request by johncarmack1984**
Was able to enlist some aid from @oscartbeaumont regarding [updated specta syntax for some of the functionality in TauRPC](https://github.com/specta-rs/specta/issues/291)... our team has had this updated fork running in our repo for a couple of weeks now, and are experiencing zero issues. :) Would love to contribute back to TauRPC, this library is helping our team so much! 

Please let me know if there are any changes you would like to see or any other procedure you'd like me to follow for this pull request to be considered for inclusion. :)  Thanks @MatsDK!

### CODE REVIEW

File: taurpc/src/export.rs [hunk 1]
```diff
@@ -43,15 +47,26 @@ pub(super) fn export_types(
         std::fs::create_dir_all(parent).unwrap();
     }
 
-    specta::export::ts(&export_path).unwrap();
+    specta_util::export()
+        .export_to(specta_typescript::Typescript::default()
+        .bigint(BigIntExportBehavior::String)
+        .formatter(specta_typescript::formatter::biome), path)
```
Comment (oscartbeaumont) [inline taurpc/src/export.rs:53]: I'm not sure these are great defaults for a library.

It would require the user to have Biome installed and your opt-ing into `BigIntExportBehavior::String` which is not Serde's default behaviour so your end-users are going to have to override number types to make the runtime types match what is exported.

It would probally be a bigger undertaking but I think it might make sense for the `Typescript` instance to be configured and parsed in by the user. This is the approach i'm using for [Tauri Specta](https://github.com/specta-rs/tauri-specta/blob/12e6774bae8498f037fda19e38eb92a3608aa27c/examples/app/src-tauri/src/main.rs#L135) and hoping to for rspc to.

Comment (johncarmack1984) [reply taurpc/src/export.rs]: Thanks @oscartbeaumont! I've updated the export to 

```rust
    specta_util::export()
        .export_to(specta_typescript::Typescript::default(), path)
        .unwrap();
```

**Comment by MatsDK** [comment]
Hey, thank you very much for doing this. Is it possible to also update the versions in the example application. Otherwise this PR look great. I will create issue for the typescript export options.

I would love to get your input on how `TauRPC` could improve error handling. Do you think we should throw the errors like we're doing now, or should we always return a  `Result` type to the frontend.

**Comment by johncarmack1984** [comment]
> Is it possible to also update the versions in the example application. 

Definitely! Done and pushed to the PR. :) Fixed my formatting oversight that borked this morning's CI run too... sorry about that!

> I would love to get your input on how `TauRPC` could improve error handling. Do you think we should throw the errors like we're doing now, or should we always return a `Result` type to the frontend.

Oh dang, good question... I think the current implementation is working great, personally. :) We're using the error responses to trigger toast notifications on frontend, so it's nice that with TauRPC the distinction is already made and we don't have to parse it ourselves. BUT if there's a strong case for `Result` responses only, I'm sure we can factor our implementation to accomodate changing it. :)

File: taurpc/src/export.rs [hunk 2]
```diff
@@ -43,15 +45,24 @@ pub(super) fn export_types(
...
 
-    file.write_all(format!("const ARGS_MAP = {}", args_map).as_bytes())
+    let args_hash: HashMap<String, String> = serde_json::from_str(&args_map).unwrap();
+    let args_entries: String = args_hash
+        .iter()
+        .map(|(k, v)| format!("'{}':'{}'", k, v))
+        .join(", ");
+    let router_args = format!("{{{}}}", args_entries);
+
+    file.write_all(format!("const ARGS_MAP = {}", router_args).as_bytes())
```
Comment (MatsDK) [inline taurpc/src/export.rs:65]: What is the reason for this change?

Comment (johncarmack1984) [reply taurpc/src/export.rs:65]: The args object was coming in with escape characters otherwise, ie `‘{\”events\”:\”{}\”}’`

Comment (MatsDK) [reply taurpc/src/export.rs:65]: Oh yeah, I see

**Comment by MatsDK** [comment]
Published with version `taurpc@0.3.1`
